### DISCUSSION TITLE
Power Control with pid controller to prevent inverter trip on 253V

### DISCUSSION THREAD

**Discussion by AlexanderKas** [binsentsu/home-assistant-solaredge-modbus#169] [Show and tell]
I use the 'new' Power Control feature combined with a [pid controller](https://github.com/soloam/ha-pid-controller) to prevent the inverter to trip on the voltage > 253V. Still busy tuning, but a great feature! 

Is their enyone doing the same thing?

**Comment by jovana** [comment]
I have the same issue on throttled inverters when voltage is peaking to 253v. 

Not sure how you are managing down the voltage using this tool. 

Now I switch on heavy devices to take the energy directly instead of export to grid (charge car, charge home battery, additional heating of the waterboiler). 

It slow down the voltage peak. But it's not perfect still.

**Reply by AlexanderKas** [reply]
Practice and theory show that decreasing the delivered power (Watt) will also decrease the voltage. The inverter has the possibility to limit the power with one value or I even think you can limit the inverter in 5 or 6 steps based on the voltage precentage, but you can also use the Power Control feature with a dynamic controller through modbus (in Home Assistant) to decrease the power based on the actual voltage, also read from the inverter.

**Comment by dnamat88** [comment]
i follow your update. i need the same feature like a pid to reduce power but without cutting huge power production. stay in touch please
